src/Microsoft.OData.ModelBuilder/Vocabularies/CapabilitiesVocabularyExtensionMethods.cs

```diff
@@ -0,0 +1,500 @@
+        {
+            _ = model ?? throw Error.ArgumentNull(nameof(model));
+            if (target == null || configuration == null)
+            {
+                return;
+            }
+
+            var restrictions = configuration.OperationRestrictions;
+            if (restrictions == null)
+            {
+                return;
+            }
+
+            var properties = new List<IEdmPropertyConstructor>();
+            if (restrictions.FilterSegmentSupported.HasValue)
+            {
+                properties.Add(new EdmPropertyConstructor(Constants.OperationRestrictions.FilterSegmentSupported, new EdmBooleanConstant(restrictions.FilterSegmentSupported.Value)));
+            }
+
+            var permissions = restrictions
+                .Permissions
+                .Select(ConvertToRecordExpression)
+                .ToArray();
+
+            properties.Add(new EdmPropertyConstructor(Constants.OperationRestrictions.Permissions, new EdmCollectionExpression(permissions)));
+
+            model.SetVocabularyAnnotation(target, properties, Constants.OperationRestrictions.Term);
+        }
+
+        private static void SetInsertRestrictionsAnnotations(this EdmModel model, EdmEntitySet target, InsertRestrictionsType restrictions)
+        {
+            if (restrictions == null || !restrictions.Insertable.HasValue)
+            {
+                return;
+            }
+
+            var properties = new List<IEdmPropertyConstructor>();
+            properties.Add(new EdmPropertyConstructor(Constants.InsertRestrictions.Insertable, new EdmBooleanConstant(restrictions.Insertable.Value)));
+
+            if (!String.IsNullOrWhiteSpace(restrictions.Description))
+            {
+                properties.Add(new EdmPropertyConstructor(Constants.InsertRestrictions.Description, new EdmStringConstant(restrictions.Description)));
+            }
+
+            if (!String.IsNullOrWhiteSpace(restrictions.LongDescription))
+            {
+                properties.Add(new EdmPropertyConstructor(Constants.InsertRestrictions.LongDescription, new EdmStringConstant(restrictions.LongDescription)));
+            }
+
+            var permissions = restrictions
+                .Permissions
+                .Select(ConvertToRecordExpression)
+                .ToArray();
+
+            properties.Add(new EdmPropertyConstructor(Constants.InsertRestrictions.Permissions, new EdmCollectionExpression(permissions)));
+
+            model.SetVocabularyAnnotation(target, properties, Constants.InsertRestrictions.Term);
+        }
+
+        private static void SetUpdateRestrictionsAnnotations(this EdmModel model, EdmEntitySet target, UpdateRestrictionsType restrictions)
+        {
+            if (restrictions == null || !restrictions.Updatable.HasValue || !restrictions.Upsertable.HasValue || !restrictions.DeltaUpdateSupported.HasValue)
```

Comment:
Shouldn't these be `&&` instead of `||`?
```suggestion
            if (restrictions == null || (!restrictions.Updatable.HasValue && !restrictions.Upsertable.HasValue && !restrictions.DeltaUpdateSupported.HasValue)
```